### ISSUE TITLE
In vscode, give ter-indent an inch, it demands a mile (inside class/interface)

### ISSUE THREAD

**Issue by ethanova** [Microsoft/vscode-tslint#230]
Note: I think this might be some kind of problem with vscode-tslint and not tslint-eslint-rules because when I run `tslint -c ./tslint.json ./src/*` it doesn't return these errors, even though the editor is showing these as present. Given that, I'm assuming I should post the bug here but if I'm off let me know.

Here's the code:
```
export class ContactCreatedEvent extends MutatingEventBase<ContactPayload, BusinessContext> implements MutatingEvent<ContactPayload, BusinessContext> {
    public static type = 'Contact_Created';
    public static version = 0;
    sequence: number;

    constructor(params: MutatingEventParams<ContactPayload>) {
      super(params);
    }

    apply(context: BusinessContext): BusinessContext {
      context.created = true;
      context.sequence = this.sequence;
      return context;
    }
}
```

Inside es6 classes and interfaces, ter-indent/tslint refuse to indent correctly. Every child line inside the class gives the error `[tslint] Expected indentation of 4 spaces but found 2. (ter-indent)
` and if I concede and give it another 2 spaces it says `[tslint] Expected indentation of 6 spaces but found 4. (ter-indent)` Gives the same issue with tabs. 

It seems to be classes/interfaces with more than one line inside as this doesn't produce the error on the one line inside:
```
export interface ICreateContact extends CQRSCommand<IContactPayload, BusinessContext, OperationalContext> {
  payload: IContactPayload;
}
```
but this does:
```
export interface ICreateContact extends CQRSCommand<IContactPayload, BusinessContext, OperationalContext> {
  payload: IContactPayload;
  test: string;
}
```
Neither of these produce the error:
```
export function test() {
  return null;
}

export function test() {
  console.log('do all the stuff');
  return null;
}
```

Relevant tslint.json
```
"indent": [
    true,
    "spaces",
    2
],
"ter-indent": [
    true,
    2,
    { "SwitchCase": 1 }
],
```

Versions:
- VSCode 1.13.1
- vscode-tslint 0.16.0
- tslint 5.4.3
- typescript 2.4.1
- Node 8.1.3
- Npm 5.1.0
- Windows 10

Let me know what other information or debugging steps I can provide to help!

**Comment by egamma** [comment]
@ethanova when you run tslint from the command line, are you sure that the tslint-eslint-rules are applied? Where do you define the rules directory, inside the tslint.json?

**Comment by ethanova** [comment]
Well, I was assuming that `tslint -c ./tslint.json ./src/*` was the correct way to pass them in, and it was showing other errors that the editor was as well. I've since cleared my console so I can't prove to myself that I didn't see these errors but I just ran it and they did come up, from that same command, this time. So I'll close this and move it to the tslint-eslint-rules repo [here](https://github.com/buzinas/tslint-eslint-rules/issues/244)

**Comment by ethanova** [comment]
So coming back to this, I discovered that I needed to run `tslint -c ./tslint.json ./src/**` with the extra * to get it to run over all files. Running that directly in the console gives me this:
```
ERROR: src/v0/commands/commands.ts[1, 79]: A space is required before '}'
ERROR: src/v0/domainRules/address.ts[12, 2]: file should end with a newline
ERROR: src/v0/events/index.ts[3, 60]: file should end with a newline
ERROR: src/v0/events/index.ts[1, 52]: Missing semicolon
ERROR: src/v0/events/index.ts[2, 60]: Missing semicolon
ERROR: src/v0/events/index.ts[3, 60]: Missing semicolon
```
But in that same directory, events, ContactCreatedEvent.ts gets the `[tslint] Expected indentation of 4 spaces but found 2. (ter-indent)` error in the IDE. So maybe it is a vscode-tslint bug?

**Comment by sbking** [comment]
@egamma @ethanova 

I have this same issue. Here is my tslint.json:

```
{
  "extends": [
    "tslint-config-standard",
    "tslint-react"
  ]
}
```

And my tsconfig.json:

```
{
  "compilerOptions": {
    "allowJs": true,
    "baseUrl": "src",
    "outDir": "out",
    "module": "esnext",
    "moduleResolution": "node",
    "allowSyntheticDefaultImports": true,
    "target": "esnext",
    "lib": ["esnext", "dom"],
    "jsx": "react-native",
    "noImplicitAny": true,
    "strictNullChecks": true,
    "noUnusedLocals": true,
    "noUnusedParameters": true,
    "experimentalDecorators": true,
    "plugins": [
      { "name": "tslint-language-service" }
    ]
  },
  "exclude": [
    "node_modules",
    "out",
    "**/*.test.ts",
    "**/*.test.js"
  ]
}
```

This is happening to me inside an interface. The error is NOT reported when I run tslint with:

```
./node_modules/.bin/tslint -p ./tsconfig.json -c ./tslint.json
```

However, the above command does report other rule violations from the config presets, such as unnecessary semicolons. The only place I ever get this issue is in Visual Studio Code, when using the tslint-language-service plugin in my tsconfig. I do not get this issue when using the VSCode tslint plugin without tslint-language-service.
